static/app/views/settings/components/dataScrubbing/modals/form/index.tsx

```diff
@@ -208,15 +212,31 @@ class Form extends Component<Props<Values, KeysOfUnion<Values>>, State> {
               required
               showHelpInTooltip
             >
-              <RegularExpression
-                type="text"
-                name="pattern"
-                placeholder={t('[a-zA-Z0-9]+')}
-                onChange={this.handleChange('pattern')}
-                value={values.pattern}
-                onBlur={onValidate('pattern')}
-                id="regex-matches"
-              />
+              <Flex gap="md" direction="column">
+                <RegularExpression
+                  type="text"
+                  name="pattern"
+                  placeholder={t('[a-zA-Z0-9]+')}
+                  onChange={this.handleChange('pattern')}
+                  value={values.pattern}
+                  onBlur={onValidate('pattern')}
+                  id="regex-matches"
+                />
+                <Flex gap="md" align="center">
+                  <Checkbox
+                    id="replace-captured"
+                    name="replaceCaptured"
+                    checked={values.replaceCaptured === 'true'}
+                    disabled={!hasCaptureGroups(values.pattern)}
+                    onChange={e =>
+                      onChange('replaceCaptured', e.target.checked.toString())
```

Comment:
It's unfortunate that `replaceCaptured` has to be a string, but it is forced by the following type:

https://github.com/getsentry/sentry/blob/f978f3603d8db387285875c75c6a7cf859a8ff39/static/app/views/settings/components/dataScrubbing/types.tsx#L98